.github/workflows/publish-crates.yml

```diff
@@ -0,0 +1,63 @@
+# This workflow is used for publish crate to crates.io. The workflow will use
+# environment "Publish to crates.io" which needs approval to access the `CARGO_REGISTRY_TOKEN`
+
+name: Publish Crates
+
+on:
+  # The workflow can be triggered or by creating new release: please ensure the tag name used for the
+  # release follow this format: '${crate name}_v{version number}', although in this workflow
+  # only ${crate name} is used, but {version number} is still needed for a good tag name.
+  release:
+    types: [created]
+  # The workflow can be also manually triggered by `workflow_dispatch`, see more
+  # from: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
+  workflow_dispatch:
+    inputs:
+      crate_name:
+        description: 'Name of crate to be published'
+        required: true
+        type: string
+permissions:
+  contents: read
+
+jobs:
+  crate_publish:
+    environment: "Publish to crates.io"
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
+            override: true
+      - name: Install build dependencies
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
```

Comment:
I'm a little bit concerned about the apt repository keys here, though not majorly concerned. If we're downloading the keys every time, then this opens us up to an attack where an attacker gets control of the LLVM or 01.org web servers, and replaces the apt key and apt repository contents with malicious packages. This would likely be noticed fairly quickly by someone (not necessarily Fortanix, but the community at large), but since this is an automated job, we might download and run a malicious payload before anyone notices and fixes it.

The alternative would be for us to download the apt keys once, check them in to the repo, and only allow those keys to be used. The downside to this approach is that we would have to manually update the apt keys when they change. These keys rarely change, so this isn't going to break us very often.

Reply:
Updated script as suggested